src/components/dockerRegistry/Docker.tsx

```diff
@@ -1444,6 +1446,59 @@ const DockerForm = ({
         return renderRegistryCredentialsAutoInjectToClustersComponent()
     }
 
+    const CreatableComponents = useMemo(
+        () => ({
+            DropdownIndicator: null,
+            ClearIndicator,
+            MultiValueRemove,
+            MultiValueContainer: (props) => <MultiValueChipContainer {...props} />,
+        }),
+        [],
+    )
+
+    const handleCreatableKeyDown: KeyboardEventHandler<HTMLDivElement> = (event) => {
+        if (!customState.repositoryList.inputValue) {
+            return
+        }
+
+        switch (event.key) {
+            case 'Enter':
+            case 'Tab':
+            case ' ': // Space
+            case ',':
+                setCustomState((prev) => ({
+                    ...prev,
+                    repositoryList: {
+                        error: '',
+                        inputValue: '',
+                        value: [
+                            ...prev.repositoryList.value,
+                            {
+                                label: customState.repositoryList.inputValue,
+                                value: customState.repositoryList.inputValue,
+                            },
+                        ],
+                    },
+                }))
+                event.preventDefault()
```

Comment:
- We should set the trimmed value in the chip
- Create a common handler for updating the state that can be reused in blur and change

Reply:
trim won't be required because as soon as tab or space is pressed a tag is created